src/whylogs/app/logger.py

```diff
@@ -75,45 +80,82 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
 
+
+    @property
+    def profile(self,):
+        """
+        :return: the last backing dataset profile
+        :rtype: DatasetProfile
+        """
+        return self._profiles[-1]["full_profile"]
+
     @property
-    def profile(self):
+    def segmented_profiles(self,)->DatasetProfile:
         """
         :return: the last backing dataset profile
         :rtype: DatasetProfile
         """
-        return self._profiles[-1]
+        
+        return self._profiles[-1]["segmented_profiles"]
+
+    def set_segments(self, segments: Union[List[List[Dict[str,str]]],List[str]]) -> None:
+        if segments:
+            if all(isinstance(elem, str) for elem in segments):
+                self.segment_type = "keys"
+                self.segments= segments
+            else:
+                self.segments=segments
+                self.segment_type = "set"
+        else:
+            self.segments=None
+            self.segment_type=None
+
+    def intialize_profiles(self, 
+        dataset_timestamp: Optional[datetime.datetime] = datetime.datetime.now(datetime.timezone.utc)) -> None:
+        
 
+        full_profile=None
+        if self.full_profile_check():
+            full_profile = DatasetProfile(
+                                    self.dataset_name,
+                                    dataset_timestamp=dataset_timestamp,
+                                    session_timestamp=self.session_timestamp,
+                                    tags=self.tags,
+                                    metadata=self.metadata,
+                                    session_id=self.session_id
+                                )
+
+        self._profiles.append({"full_profile": full_profile, "segmented_profiles" : {}})
```

Comment:
any reason we're not defining an explicit type but rather using a `dict` to hold data?

Reply:
no, more of an implement first, refactor later. Once we settle on the API we can create the data structures.